pennylane/gradients/hamiltonian_gradient.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a gradient recipe for the coefficients of Hamiltonians."""
+import pennylane as qml
+
+
+def hamiltonian_grad(tape, idx, params=None):
+    t_idx = list(tape.trainable_params)[idx]
+    op = tape._par_info[t_idx]["op"]
+    p_idx = tape._par_info[t_idx]["p_idx"]
+
+    new_tape = tape.copy(copy_operations=True, tape_cls=qml.tape.QuantumTape)
+    new_tape.set_parameters(params=params)
```

Comment:
@josh146 just FYI, I think this is needed in the hamiltonian grad function, else the original tape parameters are used, and they are not numpy arrays!

Reply:
I am so puzzled how your prototype worked without this fix!

Reply:
🤔 Something to do with `JacobianTape`?